src/Instrumentation/MySqli/src/MySqliTracker.php

```diff
@@ -0,0 +1,214 @@
+        $this->mySqliToAttributes[$mysqli][$attribute] = $value;
+    }
+
+    public function getMySqliAttributes(mysqli $mysqli) : array
+    {
+        return $this->mySqliToAttributes[$mysqli] ?? [];
+    }
+
+    public function trackMySqliFromStatement(mysqli $mysqli, mysqli_stmt $mysqli_stmt)
+    {
+        $this->statementToMySqli[$mysqli_stmt] = WeakReference::create($mysqli);
+    }
+
+    public function getMySqliFromStatement(mysqli_stmt $mysqli_stmt) : ?mysqli
+    {
+        return ($this->statementToMySqli[$mysqli_stmt] ?? null)?->get();
+        ;
+    }
+
+    public function getMySqliAttributesFromStatement(mysqli_stmt $stmt) : array
+    {
+        $mysqli = ($this->statementToMySqli[$stmt] ?? null)?->get();
+        if (!$mysqli) {
+            return [];
+        }
+
+        return $this->getMySqliAttributes($mysqli);
+    }
+
+    public function addStatementAttribute(mysqli_stmt $stmt, string $attribute, bool|int|float|string|array|null $value)
+    {
+        if (!$this->statementAttributes->offsetExists($stmt)) {
+            $this->statementAttributes[$stmt] = [];
+        }
+        $this->statementAttributes[$stmt][$attribute] = $value;
+    }
+
+    public function getStatementAttributes(mysqli_stmt $stmt) : array
+    {
+        if (!$this->statementAttributes->offsetExists($stmt)) {
+            return [];
+        }
+
+        return $this->statementAttributes[$stmt];
+    }
+
+    public function trackStatementSpan(mysqli_stmt $stmt, SpanContextInterface $spanContext)
+    {
+        $this->statementSpan[$stmt] = WeakReference::create($spanContext);
+    }
+
+    public function getStatementSpan(mysqli_stmt $stmt) : ?SpanContextInterface
+    {
+        if (!$this->statementSpan->offsetExists($stmt)) {
+            return null;
+        }
+
+        return $this->statementSpan[$stmt]->get();
+    }
+
+    public function trackMysqliSpan(mysqli $mysqli, SpanContextInterface $spanContext)
+    {
+        $this->mySqliSpan[$mysqli] = WeakReference::create($spanContext);
+    }
+
+    public function getMySqliSpan(mysqli $mysqli) : ?SpanContextInterface
+    {
+        if (!$this->mySqliSpan->offsetExists($mysqli)) {
+            return null;
+        }
+
+        return $this->mySqliSpan[$mysqli]->get();
+    }
+
+    public function trackMySqliTransaction(mysqli $mysqli, SpanContextInterface $spanContext)
+    {
+        $this->mySqliTransaction[$mysqli] = WeakReference::create($spanContext);
+    }
+
+    public function getMySqliTransaction(mysqli $mysqli) : ?SpanContextInterface
+    {
+        if (!$this->mySqliTransaction->offsetExists($mysqli)) {
+            return null;
+        }
+
+        return $this->mySqliTransaction[$mysqli]->get();
+    }
+
+    public function untrackMySqliTransaction(mysqli $mysqli)
+    {
+        if ($this->mySqliTransaction->offsetExists($mysqli)) {
+            unset($this->mySqliTransaction[$mysqli]);
+        }
+    }
+
+    private function splitQueries(string $sql)
```

Comment:
Is it possible to add a test that covers the query splitting?

Reply:
Sure thing